File: frontend/src/services/employeeService.js
import api from './api'

export const employeeService = {
  getEmployees: async (params = {}) => {
    const response = await api.get('/api/employees', { params })
    return response.data
  },

  getEmployeeById: async (id) => {
    const response = await api.get(`/api/employees/${id}`)
    return response.data
  },

  createEmployee: async (employeeData) => {
    const response = await api.post('/api/employees', employeeData)
    return response.data
  },

  updateEmployee: async (id, employeeData) => {
    const response = await api.put(`/api/employees/${id}`, employeeData)
    return response.data
  },

  deleteEmployee: async (id) => {
    const response = await api.delete(`/api/employees/${id}`)
    return response.data
  },

  getEmployeeLeaveBalances: async (id, year) => {
    const response = await api.get(`/api/employees/${id}/leave-balances`, {
      params: year ? { year } : {}
    })
    return response.data
  },

  getTeam: async (managerId) => {
    const response = await api.get(`/api/employees/manager/${managerId}/team`)
    return response.data
  }
}

export default employeeService

File: backend/src/main/java/com/erp/service/EmployeeService.java
package com.erp.service;

import com.erp.dto.request.EmployeeCreateDto;
import com.erp.dto.request.EmployeeUpdateDto;
import com.erp.dto.response.EmployeeResponseDto;
import com.erp.dto.response.LeaveBalanceDto;
import com.erp.dto.response.PaginatedResponse;
import com.erp.enums.EmployeeStatus;
import org.springframework.data.domain.Pageable;

import java.util.List;

public interface EmployeeService {
    PaginatedResponse<EmployeeResponseDto> getEmployees(
        String search, Long departmentId, EmployeeStatus status, String designation, Pageable pageable
    );

    EmployeeResponseDto getEmployeeById(Long id);

    EmployeeResponseDto getEmployeeByEmail(String email);

    EmployeeResponseDto createEmployee(EmployeeCreateDto dto);

    EmployeeResponseDto updateEmployee(Long id, EmployeeUpdateDto dto);

    void deleteEmployee(Long id);

    List<LeaveBalanceDto> getEmployeeLeaveBalances(Long employeeId, Integer year);

    List<EmployeeResponseDto> getSubordinates(Long managerId);
}
